File: backend/src/main/java/com/ll/hotel/domain/hotel/hotel/dto/GetAllHotelOptionsResponse.java
package com.ll.hotel.domain.hotel.hotel.dto;

import com.ll.hotel.domain.hotel.option.entity.HotelOption;
import java.util.List;
import java.util.Set;
import java.util.stream.Collectors;

public record GetAllHotelOptionsResponse(
        Set<String> hotelOptions
) {
    public GetAllHotelOptionsResponse(List<HotelOption> hotelOptions) {
        this(
                hotelOptions.stream()
                        .map(HotelOption::getName)
                        .collect(Collectors.toSet())
        );
    }
}

File: backend/src/main/java/com/ll/hotel/domain/hotel/hotel/dto/GetAllHotelOptionsResponse.kt
package com.ll.hotel.domain.hotel.hotel.dto

import com.ll.hotel.domain.hotel.option.entity.HotelOption

data class GetAllHotelOptionsResponse(
    val hotelOptions: Set<String>
) {
    constructor(hotelOptions: List<HotelOption>) : this(
        hotelOptions.map { it.name }.toSet()
    )
}